lms/views/api/sync.py

```diff
@@ -0,0 +1,66 @@
+from marshmallow import Schema
+from pyramid.view import view_config
+from webargs import fields
+
+from lms.security import Permissions
+from lms.validation._base import PyramidRequestSchema
+
+
+class APISyncSchema(PyramidRequestSchema):
+    class LMS(Schema):
+        product = fields.Str(required=True)
+
+    class Course(Schema):
+        context_id = fields.Str(required=True)
+
+    class Assignment(Schema):
+        group_set_id = fields.Str(required=False, allow_none=True)
+
+    lms = fields.Nested(LMS, required=True)
+    course = fields.Nested(Course, required=True)
+    assignment = fields.Nested(Assignment, required=True)
+
+    gradingStudentId = fields.Str(required=False, allow_none=True)
```

Comment:
I agree with this. The casing will standout in the FE too but there it will be part of an external API and it will be consistent with the rest of the API fields.

That change needs backwards compatibility precautions anyway (as the FE it's sending it in all version) so I rather tackle that on a more focused PR.